crates/client/rpc/src/test_utils.rs

```diff
@@ -13,11 +13,41 @@ use dp_state_update::{
     StorageEntry,
 };
 use dp_transactions::{InvokeTransaction, InvokeTransactionV0, Transaction};
+use jsonrpsee::core::{async_trait, RpcResult};
 use rstest::fixture;
-use starknet_core::types::Felt;
+use starknet_core::types::{
+    BroadcastedDeclareTransaction, BroadcastedDeployAccountTransaction, BroadcastedInvokeTransaction,
+    DeclareTransactionResult, DeployAccountTransactionResult, Felt, InvokeTransactionResult,
+};
 use std::sync::Arc;
 
-use crate::{providers::TestTransactionProvider, Starknet};
+use crate::{providers::AddTransactionProvider, Starknet};
+
+#[cfg(test)]
+pub struct TestTransactionProvider;
+
+#[cfg(test)]
+#[async_trait]
+impl AddTransactionProvider for TestTransactionProvider {
```

Comment:
i pulled in the automock crate in #238 for this kind of thing, what do you think?

Reply:
way better yup!